navigation/perfect_memory_base.h

```diff
@@ -79,30 +79,8 @@ class PerfectMemoryBase
     template<class... Ts>
     std::vector<std::vector<float>> getImageDifferences(Ts &&... args) const
     {
-        const size_t numSnapshots = getNumSnapshots();
-        assert(numSnapshots > 0);
-
-        Rotater rotater(getUnwrapResolution(), getMaskImage(), std::forward<Ts>(args)...);
-
-        // Create vector to store RIDF values
-        std::vector<std::vector<float>> differences(numSnapshots);
-        for (auto &d : differences) {
-            d.resize(rotater.max());
-        }
-
-        // Scan across image columns
-        rotater.rotate(
-            [this, &differences, numSnapshots](const cv::Mat &fr, const cv::Mat &mask, size_t i)
-            {
-                // Loop through snapshots
-                for (size_t s = 0; s < numSnapshots; s++) {
-                    // Calculate difference
-                    const auto diff = calcSnapshotDifference(fr, mask, s);
-                    differences[s][i] = diff;
-                }
-            });
-
-        return differences;
+        calcImageDifferences(std::forward<Ts>(args)...);
+        return m_Differences;
```

Comment:
This function should probably now return a const reference - this will still copy as it won't move a member's data out from under it